builder/dockerfile/internals.go

```diff
@@ -137,6 +171,53 @@ func (b *Builder) performCopy(state *dispatchState, inst copyInstruction) error
 	return b.exportImage(state, imageMount, runConfigWithCommentCmd)
 }
 
+func parseChownFlag(flChown string) (string, string) {
```

Comment:
I don't think the complexity changes much, it's just being moved somewhere more appropriate, so there's less duplicate code to contain.